src/routers/observations.py

```
@@ -0,0 +1,171 @@
+            select(models.Document)
+            .where(models.Document.id == observation_id)
+            .where(models.Document.workspace_name == workspace_id)
+            .where(models.Document.session_name == session_id)
+        )
+        result = await db.execute(stmt)
+        document = result.scalar_one_or_none()
+
+        if document is None:
+            raise ResourceNotFoundException(
+                f"Observation {observation_id} not found in session {session_id}"
+            )
```

Ruff (0.14.2)

Abstract `raise` to an inner function

(TRY301)